javagen/src/main/java/com/azure/autorest/util/ModelExampleUtil.java

```diff
@@ -178,8 +178,6 @@ public static ExampleNode parseNode(IType type, IType wireType, Object objectVal
                 // e.g. do not throw exception, use defaultValueExpression
                 node = defaultNode(type, wireType, objectValue);
             }
-        } else if (objectValue == null) {
-            node = null;
         } else {
             node = defaultNode(type, wireType, objectValue);
```

Comment:
Is it expected that we merge this 2 case into 1?

Reply:
Good catch. I should've brought this up. 

May I ask what does null mean for ExampleNode previously? I'm guessing it means absence of node, but I wasn't able to find any null checks for returned `parseNode`. Closest I can find is that everywhere we call `parseNode`, we ensure the passed in `objectValue` is not null, e.g. https://github.com/Azure/autorest.java/blob/9b59bd39524bd50b79d04458b464d5d4a241680b/fluentgen/src/main/java/com/azure/autorest/fluent/mapper/ExampleParser.java#L479-L490
, and this behaves identical to `defaultNode`. 

In this sense, I kinda feel like null is an unexpected return value. Would like your opinion on this.

Reply:
I feel like letting downstream decide what to do with node with null value, instead of a null node.

Reply:
For some (commonly seen) cases, if the value is null, this is already filtered out, e.g. for class model.

```
                for (Object childObjectValue : elements) {
                    ExampleNode childNode = parseNode(elementType, childObjectValue);
                    node.getChildNodes().add(childNode);
                }
```

for parameter, it seems also already handled directly
```
        if (exampleValue == null) {
            if (ClassType.CONTEXT.equals(methodParameter.getClientMethodParameter().getClientType())) {
                node = new LiteralNode(ClassType.CONTEXT, "").setLiteralsValue("");
            } else {
                node = new LiteralNode(methodParameter.getClientMethodParameter().getClientType(), null);
            }
        } else {
```

I guess one remaining place would be a `null` value in JSON array...

Reply:
> I feel like letting downstream decide what to do with node with null value, instead of a null node. Added comment on this.

Yeah, make sense, as long as we've handled the known types, and what's left be Literal (basically types about primitive or datetime etc.).

Reply:
Thanks! 
```
                for (Object childObjectValue : elements) {
                    ExampleNode childNode = parseNode(elementType, childObjectValue);
                    node.getChildNodes().add(childNode);
                }
```
This seems to be the JSON array case? But yeah, we filtered out null for class model: 
https://github.com/Azure/autorest.java/blob/9b59bd39524bd50b79d04458b464d5d4a241680b/javagen/src/main/java/com/azure/autorest/util/ModelExampleUtil.java#L112

And
https://github.com/Azure/autorest.java/blob/9b59bd39524bd50b79d04458b464d5d4a241680b/javagen/src/main/java/com/azure/autorest/util/ModelExampleUtil.java#L146-L148

Then I prefer to use a default node with null value instead of a null node. 
1. We need to deal with every node, either property or parameter. For missing value, we have to determine its default value based on its client type.
2. We don't have to do null checks for node when we use it.
3. For List case, we don't have to pay extra attention about null value in example array.